src/main/java/org/mondemand/Client.java

```diff
@@ -230,11 +232,17 @@ public void finalize() {
     contextStats.clear();
 
     // shutdown all the transports
-    if (transports != null) {
-      for (Transport t : transports) {
-        try {
-          t.shutdown();
-        } catch(TransportException e) {}
+    Set<Transport> seenTransports = new HashSet<Transport>();
```

Comment:
some transports may appear more than once in map, so keep track of which ones have already been shut down.